tests/integration/test_orders_api.py

```diff
@@ -149,26 +149,36 @@ async def test_list_orders_state_invalid_state(session):
 
 @respx.mock
 @pytest.mark.asyncio
-async def test_list_orders_limit(order_descriptions, session):
+@pytest.mark.parametrize("limit,limitted_list_length", [(None, 100), (0, 102),
+                                                        (1, 1)])
+async def test_list_orders_limit(order_descriptions,
+                                 session,
+                                 limit,
+                                 limitted_list_length):
     nono_page_url = TEST_ORDERS_URL + '?page_marker=OhNoNo'
```

Comment:
@kevinlacaille I found the problem! The "next" link evaluates to `True` on all the result pages, so we have infinitely many pages and the client never breaks out of the paging loop.

```suggestion
    nono_page_url = None
```

Reply:
If you really want to see what's going on, try replacing some of your test code with this:

```python
    cl = OrdersClient(session, base_url=TEST_URL)

    # import pdb; pdb.set_trace()
    orders = await cl.list_orders(limit=limit)

    async def iter_orders(orders):
        i = 1
        # import pdb; pdb.set_trace()
        async for page in orders._get_pages():
            for item in page["orders"]:
                yield item
                i += 1
                print(f"item {i}")
                if (orders.limit and i > orders.limit) or i > 200:
                    print(f"Reached limit {i}")
                    break
            else:
                continue
            break

    assert len([o async for o in iter_orders(orders)]) == limitted_list_length
```